android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -166,7 +172,11 @@ protected void doFrameGuarded(long frameTimeNanos) {
     // Events are handled in the native modules thread in the `onEventDispatch()` method.
     // This method indirectly uses `mChoreographerCallback` which was created after event
     // registration, creating race condition
-    mUIManager.getEventDispatcher().addListener(this);
+    if (context.isBridgeless()) {
+      ((FabricUIManager) mUIManager).getEventDispatcher().addListener(this);
+    } else {
+      ((UIManagerModule) mUIManager).getEventDispatcher().addListener(this);
+    }
```

Comment:
I think we can improve this if. What if someone is in on the new architecture but with disabled bridgeless? In my opinion, the logic should say: 
```
if (new arch) { 
  use fabric 
} else { 
  use UI manager
} 
```

Reply:
> What if someone is in on the new architecture but with disabled bridgeless?

It works correctly then because we only use `FabricUIManager` on bridgeless as can be seen in your other commented line. Do you think we should change it if `UIManagerModule` is still available on new arch but without bridgeless mode?